### PR TITLE
test: connection url appstream

### PR DESCRIPTION
**Pull request by SanketD92**
Issue #, if available:
GALI-1102

Description of changes:
Added integration tests for AppStream URL verification

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully tested with your changes locally?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: addons/addon-base-raas-appstream/packages/base-raas-appstream-services/lib/plugins/aws-account-mgmt-plugin.js [hunk 1]
```diff
@@ -35,19 +36,24 @@ async function getActiveNonAppStreamEnvs({ awsAccountId }, { requestContext, con
   const indexesService = await container.find('indexesService');
 
   const indexes = await indexesService.list(requestContext);
-  const indexesIdsOfInterest = _.map(
-    _.filter(indexes, index => index.awsAccountId === awsAccountId),
-    'id',
-  );
+  const indexesOfInterest = _.filter(indexes, index => index.awsAccountId === awsAccountId);
```
Comment (nguyen102) [inline addons/addon-base-raas-appstream/packages/base-raas-appstream-services/lib/plugins/aws-account-mgmt-plugin.js:39]: Not a big deal, but for arrays, JS comes with the `filter` method already. Therefore we don't have to use lodash for this. We don't have to change it for this PR, but for future reference. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

File: main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const fetch = require('node-fetch');
```
Comment (nguyen102) [inline main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js]: `axios` is already included in `integration-tests`. Should we just continue using that?

File: main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+          applicationName: 'EC2Linux',
+          preAuthStreamingUrl: 'https://appstream2.us-east-1.aws.amazon.com/authenticate?parameters=',
+          redirectStreamingUrl:
+            'appstream2.us-east-1.aws.amazon.com/#/streaming/?reference=fleet%2Finitial-stack-1629237287942-ServiceWorkbenchFleet&app=',
+          expected: { scheme: 'ssh', name: 'SSH to Main EC2 instance' },
+        },
+      ],
+    ];
+
+    it.each(testContexts)('should return AppStream URL %s', async (_, testContext) => {
```
Comment (nguyen102) [inline main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js]: Nitpick: `should return AppStream URL: %s`

File: main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      const redirectStreamingUrl = `appstream2.us-east-1.aws.amazon.com/#/streaming/?reference=fleet%2Finitial-stack-1629237287942-ServiceWorkbenchFleet&app=${applicationName}`;
+
+      const retVal = await adminSession.resources.workspaceServiceCatalogs
+        .workspaceServiceCatalog(envId)
+        .connections()
+        .connection(connectionId)
+        .createUrl();
+
+      const response = await fetch(retVal.url);
+      const redirectResponse = response[Object.getOwnPropertySymbols(response)[1]];
```
Comment (nguyen102) [inline main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js]: I would be interested in hearing your thoughts on this. Can we make the test simpler to read by moving the checks closer to the action? Example.

```
// BUILD
const envId = '36f22de5-fefb-4f62-8ab4-e99ac4387ad4';
const connectionId = 'id-0';
const applicationName = 'Firefox';
const sagemakerUrlPrefix = 'https://basicnotebookinstance';
const preAuthStreamingUrl = 'https://appstream2.us-east-1.aws.amazon.com/authenticate?parameters=';
const redirectStreamingUrl = `appstream2.us-east-1.aws.amazon.com/#/streaming/?reference=fleet%2Finitial-stack-1629237287942-ServiceWorkbenchFleet&app=${applicationName}`;

// OPERATE
const retVal = await adminSession.resources.workspaceServiceCatalogs
    .workspaceServiceCatalog(envId)
    .connections()
    .connection(connectionId)
    .createUrl();

// CHECK
expect(retVal.appstreamDestinationUrl).toContain(sagemakerUrlPrefix);
expect(retVal.id).toEqual(connectionId);
expect(retVal.url).toContain(preAuthStreamingUrl);

// OPERATE
const response = await fetch(retVal.url);
const redirectResponse = response[Object.getOwnPropertySymbols(response)[1]];

// CHECK
expect(redirectResponse.url).toContain(redirectStreamingUrl);
  ```

File: main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      const retVal = await adminSession.resources.workspaceServiceCatalogs
+        .workspaceServiceCatalog(testContext.envId)
+        .connections()
+        .connection(testContext.connectionId)
+        .createUrl();
+
+      const response = await fetch(retVal.url);
+      const redirectResponse = response[Object.getOwnPropertySymbols(response)[1]];
+
+      expect(retVal.scheme).toContain(testContext.expected.scheme);
```
Comment (nguyen102) [inline main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js]: Same here with reorganizing the order of the checks?

File: main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  describe('Create AppStream URL', () => {
+    it('should return AppStream URL SageMaker', async () => {
+      const envId = '36f22de5-fefb-4f62-8ab4-e99ac4387ad4';
+      const connectionId = 'id-0';
+      const applicationName = 'Firefox';
+      const sagemakerUrlPrefix = 'https://basicnotebookinstance';
+      const preAuthStreamingUrl = 'https://appstream2.us-east-1.aws.amazon.com/authenticate?parameters=';
+      const redirectStreamingUrl = `appstream2.us-east-1.aws.amazon.com/#/streaming/?reference=fleet%2Finitial-stack-1629237287942-ServiceWorkbenchFleet&app=${applicationName}`;
+
+      const retVal = await adminSession.resources.workspaceServiceCatalogs
```
Comment (nguyen102) [inline main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js]: Nitpick: `retVal` => `connectionUrlResponse`

File: addons/addon-base-raas-appstream/packages/base-raas-appstream-services/lib/plugins/aws-account-mgmt-plugin.js (hunk 1 above)
Comment (SanketD92) [reply addons/addon-base-raas-appstream/packages/base-raas-appstream-services/lib/plugins/aws-account-mgmt-plugin.js:39]: True, but there's a minor performance [difference](https://www.measurethat.net/Benchmarks/Show/4355/0/arrayprototypefilter-vs-lodash-filter).

File: main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js (hunk 4 above)
Comment (SanketD92) [reply main/integration-tests/__test__/api-tests/appstream-egress-enabled/workspace-service-catalog/connections/create-url-connection.test.js]: Yup, good idea. I'll make the change.

**Review by nguyen102 — APPROVED** [review]

**Pull request merged**